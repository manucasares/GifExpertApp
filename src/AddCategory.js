import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState("");

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleInputSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories((prevCategorie) => [inputValue, ...prevCategorie]);
            setInputValue("");
        }
    };

    return (
        <form onSubmit={handleInputSubmit}>
            <p> {inputValue}</p>
            <input
                type="text"
                placeholder="Busque un GIF"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
