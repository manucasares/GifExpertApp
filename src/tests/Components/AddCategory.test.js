import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import AddCategory from "../../AddCategory";

describe("Pruebas en AddCategory", () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test("debe mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("debe de cambiar el input value ", () => {
        const input = wrapper.find("input");

        const value = "Hola mundo";

        input.simulate("change", {
            target: {
                value: value,
            },
        });

        expect(wrapper.find("p").text().trim()).toBe(value);
    });

    test("No debe de postear la informacion con submit", () => {
        wrapper.find("form").simulate("submit", {
            preventDefault() {},
        });

        expect(setCategories).not.toHaveBeenCalled();
    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        
        //1 - simular el inputChange
        const input = wrapper.find('input');
        const value = "Hola mundo";
        input.simulate('change', {
            target : {
                value : value
            }
        })

        //2 - simular el submit
        const form = wrapper.find('form')
        form.simulate('submit', {
            preventDefault(){}
        })

        //3 - setCategories se debe haber llamado
        expect(setCategories).toHaveBeenCalled();

        //4 - el valor del input debe estar ''
        expect(input.text().trim()).toBe('');




    })
    
});
