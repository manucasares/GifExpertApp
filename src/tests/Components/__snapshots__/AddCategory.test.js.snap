// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Busque un GIF"
    type="text"
    value=""
  />
</form>
`;
